furryfeet_be/furryfeet/views.py

```diff
@@ -436,4 +437,222 @@ def post(self, request):
\ No newline at end of file
+        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+
+#FeedbackListView
+class FeedbackListView(APIView):
+    def get(self, request):
+        token = request.COOKIES.get('jwt')
+
+        if not token:
+            raise AuthenticationFailed("Unauthenticated!")
+
+        try:
+            payload = jwt.decode(token, 'secret', algorithms=['HS256'])
+        except jwt.ExpiredSignatureError:
+            raise AuthenticationFailed("Unauthenticated!")
+
+        feedbacks = Feedback.objects.all()
+        serializer = FeedbackSerializer(feedbacks, many=True)
+
+        return Response(serializer.data, status=status.HTTP_200_OK)
+
+
+class FeedbackCreateView(APIView):
+    def post(self, request):
+        token = request.COOKIES.get('jwt')
+
+        if not token:
+            raise AuthenticationFailed("Unauthenticated!")
+
+        try:
+            payload = jwt.decode(token, 'secret', algorithms=['HS256'])
+        except jwt.ExpiredSignatureError:
+            raise AuthenticationFailed("Unauthenticated!")
+
+        # get the user who is posting the feedback
+        user_from = User.objects.get(id=payload['id'])
+
+        # get the user who is receiving the feedback
+        user_to_id = request.data.get('user_to')
+        user_to = User.objects.get(id=user_to_id)
+
+        # create the feedback object
+        feedback = Feedback(
+            rating=request.data.get('rating'),
+            comment=request.data.get('comment'),
+            user_from=user_from,
+            user_to=user_to
+        )
+        feedback.save()
+
+        serializer = FeedbackSerializer(feedback)
+        return Response(serializer.data, status=status.HTTP_201_CREATED)
+
+class FeedbackDetailView(APIView):
+    def get(self, request, feedback_id):
+        token = request.COOKIES.get('jwt')
+
+        if not token:
+            raise AuthenticationFailed("Unauthenticated!")
+
+        try:
+            payload = jwt.decode(token, 'secret', algorithms=['HS256'])
+        except jwt.ExpiredSignatureError:
+            raise AuthenticationFailed("Unauthenticated!")
+
+        feedback = Feedback.objects.get(pk=feedback_id)
+        serializer = FeedbackSerializer(feedback)
+
+        return Response(serializer.data, status=status.HTTP_200_OK)
+
+class AppointmentCreateView(APIView):
+    def post(self, request):
+        # get the JWT token from the request
+        token = request.COOKIES.get('jwt')
+
+        if not token:
+            raise AuthenticationFailed('Unauthenticated!')
+
+        try:
+            payload = jwt.decode(token, 'secret', algorithms=['HS256'])
+        except jwt.ExpiredSignatureError:
+            raise AuthenticationFailed('Unauthenticated!')
+
+        # get the dog walker id from request data
+        dog_walker_id = request.data.get('dog_walker')
+
+        iso_format = '%Y-%m-%d %H:%M:%S.%f'
```

Comment:
either move this iso format to the begining of the method or somewhere around line 544, since it makes no sense for it to be in the middle of nowhere